tests/basic_ref.rs

```diff
@@ -368,33 +271,41 @@ fn drop_value() {
 fn clone_map_empty() {
     let map = HashMap::<&'static str, u32>::new();
     let map = map.pin();
-    let cloned_map = map.clone(); // another ref to the same map
+    let cloned_map = map.clone();
     assert_eq!(map.len(), cloned_map.len());
-    assert_eq!(map, cloned_map);
+    assert_eq!(&map, &cloned_map);
     assert_eq!(cloned_map.len(), 0);
 }
 
 #[test]
-// Test that same values exists in both refs (original and cloned)
+// TODO: Should this test really fail?
```

Comment:
What should we do with this test?

Reply:
Ah, I think that test is _supposed_ to hold an `Arc<HashMap>`, not just a `HashMap`, and that the clone is supposed to just clone the `Arc`, not the entire `HashMap`. With that, the original test would be correct.

Reply:
If I change the "normal" (`HashMap` instead of `HashMapRef`) to use Arc, it fails with the same error.

Reply:
What error is that? Are you using `Arc::clone` to clone the `Arc<HashMap>` (not just calling `.clone()`)?

Reply:
If the original test uses `Arc`, it fails with the same error.

Reply:
Yep I'm using `Arc::clone`

Reply:
Right, but what error is that? Also, when changing it to use `Arc`, you also need to change the line that does `.clone()` to instead do `Arc::clone(&map)`.

Reply:
```rs
---- clone_map_filled stdout ----
thread 'clone_map_filled' panicked at 'assertion failed: `(left != right)`
  left: `{"FooKey": 0, "NewItem": 100, "BarKey": 10}`,
 right: `{"FooKey": 0, "NewItem": 100, "BarKey": 10}`', tests/basic.rs:491:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
That's the error

Reply:
Oh oh oh, I think I understand what you mean now. So, the original test is testing that if you clone the map, you end up with two distinct copies, and that modifying one copy does not modify the other. That test is correct. With `HashMapRef`, the semantics are different: cloning a `HashMapRef` gives you a new `HashMapRef` _to the same map_. Those are the semantics we want (I believe), so the test for ref will necessarily have to be different. Specifically, in ref, it should assert that the two clones of the map are indeed equal after the modification.

Reply:
Ahh okay. Thank you :+1: